src/dmd/mars.d

```diff
@@ -848,21 +848,107 @@ private int tryMain(size_t argc, const(char)** argv)
         fatal();
 
     // inlineScan incrementally run semantic3 of each expanded functions.
-    // So deps file generation should be moved after the inlinig stage.
-    if (global.params.moduleDeps)
+    // So deps file generation should be moved after the inlining stage.
+    if (global.params.moduleDeps || global.params.doMakeDeps)
```

Comment:
You probably don't want to reuse `-deps` functionality as it is no longer useful for `make`. Analysing code via `semantic3OnDependencies` also reports local imports that are not used when *compiling* the code without the `-deps` option. It can also slow down compilation considerably. See discussion in https://github.com/dlang/dmd/pull/6748.

Reply:
That was insightful, thank you. I've moved the `semantic3OnDependencies` call to be done only for the old `-deps` switch.